### PR TITLE
Add search.index_format into the serialized config (#2165)

### LINKED ISSUE #2165: [bug] config.search.index_format not accessible from templates.
# Bug Report

I am in the process of [refactoring the Abridge theme](https://github.com/Jieiku/abridge/issues/105).
One of the things I am doing is switching over to the [new json index format](https://github.com/getzola/zola/pull/1998)

When using the new json index format, you use a different search.js then you would with the regular js index. (this new version fetches the json file on interaction)

because of this I need to ensure or check that the right configuration is set.

```html
  {%- if config.search.index_format == "elasticlunr_json" %}
  <script defer src="{{ get_url(path='search_json.js', trailing_slash=false, cachebust=true) | safe }}"{%- if config.extra.integrity | default(value=true) %} integrity="sha384-{{ get_hash(path='search_json.js', sha_type=384, base64=true) | safe }}"{%- endif %}></script>
  {%- else %}
  <script defer src="{{ get_url(path='search_js.js', trailing_slash=false, cachebust=true) | safe }}"{%- if config.extra.integrity | default(value=true) %} integrity="sha384-{{ get_hash(path='search_js.js', sha_type=384, base64=true) | safe }}"{%- endif %}></script>
  {%- endif %}
```

## Environment

Zola version: 0.17.1

## Expected Behavior
I need for `config.search.index_format` to be accessible.

OR maybe `build_search_index = true` could also be `build_search_index = "elasticlunr_json"`  then there would not even be the need for the additional `index_format` field.

## Current Behavior
Currently you cannot check the value of `config.search.index_format` from templates

### PR DESCRIPTION
**Pull request by Raymi306**
**IMPORTANT: Please do not create a Pull Request adding a new feature without discussing it first.**

The place to discuss new features is the forum: <https://zola.discourse.group/>
If you want to add a new feature, please open a thread there first in the feature requests section.

Sanity check:

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes
(Delete or ignore this section for documentation changes)

* [x] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [x] Have you created/updated the relevant documentation page(s)?
---
Fixes #2165 

If there is interest in continuing with this PR, I will update documentation

### CODE REVIEW

**Comment by Raymi306** [comment]
@Jieiku This should enable you to do what you need

**Comment by Jieiku** [comment]
> @Jieiku This should enable you to do what you need

I have a few open tickets, which one would this fix? would love to test it out.

**Comment by Jieiku** [comment]
Thanks! Going to give it a try!

**Comment by Keats** [comment]
I would start with just the index format serialized and then we can see if we need more. The rest feels like config that you don't need in the templates.

**Comment by Raymi306** [comment]
> I would start with just the index format serialized and then we can see if we need more. The rest feels like config that you don't need in the templates.

This makes sense.

I looked in the docs for places where I want to mention this variable being available, perhaps [here](https://www.getzola.org/documentation/templates/overview/) in templates/overview?

Would you like any tests for this?

File: components/config/src/config/mod.rs [hunk 1]
```diff
@@ -109,6 +109,7 @@ pub struct SerializedConfig<'a> {
     build_search_index: bool,
     extra: &'a HashMap<String, Toml>,
     markdown: &'a markup::Markdown,
+    search_index_format: &'a search::IndexFormat,
```
Comment (Keats) [inline components/config/src/config/mod.rs]: You can create a new struct (have a look at how we do it for page and sections) for the data serialized so if we add more items it's nicely contained.

**Comment by Keats** [comment]
> I looked in the docs for places where I want to mention this variable being available, perhaps [here](https://www.getzola.org/documentation/templates/overview/) in templates/overview?

No need I think, it's going to be part of the config which is documented.

> Would you like any tests for this?

No need for just serialization

**Review by Keats — APPROVED** [review]

**Comment by Keats** [comment]
Thanks!

**Pull request merged**